src/core/wee-string.c

```diff
@@ -224,6 +225,10 @@ string_repeat (const char *string, int count)
         return strdup (string);
 
     length_string = strlen (string);
+
+    if (INT_MAX / length_string <= count)
```

Comment:
For better readability, I would reverse the condition, like this:

```c
if (count >= INT_MAX / length_string)
```
And this error case must be tested in the unit test there: https://github.com/weechat/weechat/blob/master/tests/unit/core/test-core-string.cpp#L300
Thanks

Reply:
Fixed. I have also added a unit test. Thank you for pointing out exactly where to place it. I have used INT_MAX there as well, although I doubt that weechat will run on a system where sizeof(int) is not 4 bytes.

If you prefer a const integer value, please let me know.